src/main/kotlin/sliv/tool/scene/view/virtualizedfx/VirtualizedFXGridProvider.kt

```diff
@@ -5,26 +5,45 @@ import io.github.palexdev.mfxcore.collections.ObservableGrid
 import io.github.palexdev.virtualizedfx.enums.ScrollPaneEnums
 import io.github.palexdev.virtualizedfx.grid.VirtualGrid
 import io.github.palexdev.virtualizedfx.utils.VSPUtils
-import javafx.scene.Node
+import javafx.beans.property.DoubleProperty
+import javafx.geometry.Orientation
 import sliv.tool.scene.model.VisualizationFrame
 import sliv.tool.scene.view.*
+import tornadofx.onChange
 
 object VirtualizedFXGridProvider : GridProvider {
     override fun createGrid(
         data: List<VisualizationFrame>,
         columnsNumber: Int,
         cellWidth: Double,
         cellHeight: Double,
+        scale: DoubleProperty,
         cellFactory: (VisualizationFrame) -> FrameView
-    ): Node {
+    ): Grid {
         val gridData = ObservableGrid.fromList(data, columnsNumber)
         val grid = VirtualGrid(gridData) { item -> FrameViewAdapter(cellFactory(item)) }
         grid.cellSize = Size(cellWidth, cellHeight)
 
         val vsp = VSPUtils.wrap(grid)
         vsp.layoutMode = ScrollPaneEnums.LayoutMode.COMPACT
         vsp.isAutoHideBars = true
+        vsp.isSmoothScroll = true
+        VSPUtils.setVSpeed(vsp, 100.0, 100.0, 100.0)
 
-        return vsp
+        scale.onChange { newScale ->
+            // Virtual grid breaks if cell size changes when x position differs from 0.0
+            // Virtual grid always drops position to (0, 0) when cell size changes, so this solution is ok
+            grid.scrollTo(0.0, Orientation.HORIZONTAL)
```

Comment:
is scroll reseted here?

Reply:
how it will be restored?

Reply:
I reset scroll to prevent exception from VirtualGrid. It doesn't change anything, because on cell size changing VirtualScrollPane resets scroll. Position is restored in zoomGrid method (SceneView.kt).